command/server.go

```diff
@@ -237,11 +239,29 @@ func (c *ServerCommand) Run(args []string) int {
 		DefaultLeaseTTL:    config.DefaultLeaseTTL,
 		ClusterName:        config.ClusterName,
 		CacheSize:          config.CacheSize,
+		PluginDirectory:    config.PluginDirectory,
 	}
 	if dev {
 		coreConfig.DevToken = devRootTokenID
 	}
 
+	if config.PluginDirectory == "" {
```

Comment:
I'm not really a fan of this logic, for two reasons:

1) `homedir` will use `HOME` env var if possible but otherwise will start exec'ing commands (getent, shell) to try discovery; I feel like this could well be rather unexpected to the user since Vault doesn't otherwise normally exec processes

2)  Making a directory both seems unexpected as a user and it's not even clear why we're doing it in the first place. If a dir isn't provided (or doesn't exist) it seems like we just have an empty plugin catalog (outside of built-in plugins).